shoop/core/models/_contacts.py

```diff
@@ -359,7 +359,7 @@ def get_person_contact(user):
         'last_name': getattr(user, 'last_name', ''),
         'email': getattr(user, 'email', ''),
     }
-    return PersonContact.objects.get_or_create(user=user, defaults=defaults)[0]
+    return PersonContact.objects.filter(is_active=True).get_or_create(user=user, defaults=defaults)[0]
```

Comment:
> Also this will cause dublicate entry database error since the user is onetoone-field.

You're talking about the Shawn's implementation which does `.filter(...).get_or_create(...)`. Right? The snippet I posted, doesn't have that problem.

> I think the correct fix here would be to logout the user and show message that your account is deactivated.

That's not job of this function. It would be the middleware that should do that.

> I think this shouldn't never return None.

Hmm... I was about to disagree, because then it would be working differently than `get_company_contact`, but now that I looked for usages of `get_person_contact` I think you're right. It's probably wiser to always return a contact.

> Also user shouldn't be logged in without actual active contact in request.

Yeah, for Front. Login to Admin is OK without a customer in request. Preventing login to Front without active contact is implemented in `shoop.front.apps.auth.forms.EmailAuthenticationForm.confirm_login_allowed`. Though that currently does not seem to use this `get_person_contact` function, maybe it should?